hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -346,7 +346,8 @@ public void write(HoodieRecord<T> oldRecord) {
       // writing the first record. So make a copy of the record to be merged
       HoodieRecord<T> newRecord = keyToNewRecords.get(key).newInstance();
       try {
-        Option<Pair<HoodieRecord, Schema>> mergeResult = recordMerger.merge(oldRecord, oldSchema, newRecord, newSchema, props);
+        Option<Pair<HoodieRecord, Schema>> mergeResult = recordMerger.merge(
```

Comment:
We should also call `merge` API for those records which existed in `keyToNewRecords` but not existed in base file before flushed them to disk.

Reply:
I think we should just have one merge API and educate user how to implement it with custom logic.  More APIs (like in payload class) are just going to add more thinking overhead.